module/VuFind/src/VuFind/ILS/Driver/Folio.php

```diff
@@ -1090,27 +1128,52 @@ protected function performOkapiUsernamePasswordAuthentication(string $username,
 
     /**
      * Given a response from performOkapiUsernamePasswordAuthentication(),
-     * extract the token value.
+     * extract the requested cookie.
      *
-     * @param Response $response Response from performOkapiUsernamePasswordAuthentication().
+     * @param Response $response   Response from performOkapiUsernamePasswordAuthentication().
+     * @param string   $cookieName Name of the cookie to get from the response.
      *
-     * @return string
+     * @return \Laminas\Http\Header\SetCookie
      */
-    protected function extractTokenFromResponse(Response $response): string
+    protected function getCookieByName(Response $response, string $cookieName): \Laminas\Http\Header\SetCookie
     {
-        if ($this->useLegacyAuthentication()) {
-            return $response->getHeaders()->get('X-Okapi-Token')->getFieldValue();
-        }
         $folioUrl = $this->config['API']['base_url'];
         $cookies = new \Laminas\Http\Cookies();
         $cookies->addCookiesFromResponse($response, $folioUrl);
         $results = $cookies->getAllCookies();
         foreach ($results as $cookie) {
-            if ($cookie->getName() == 'folioAccessToken') {
-                return $cookie->getValue();
+            if ($cookie->getName() == $cookieName) {
+                return $cookie;
             }
         }
-        throw new \Exception('Could not find token in response');
+        throw new \Exception('Could not find ' . $cookieName . ' in response');
```

Comment:
Might not hurt to clarify this error message slightly.

```suggestion
        throw new \Exception('Could not find ' . $cookieName . ' cookie in response');
```

Reply:
Good idea, that's been updated.